system-test/search.js

```diff
@@ -67,12 +56,8 @@ describe('Search', function() {
   var INDEX_NAME = generateIndexName();
   var index = search.index(INDEX_NAME);
 
-  before(function(done) {
-    deleteAllDocuments(done);
```

Comment:
Yes, if a test that created a document timed out and the rest of our test suite finished before that test had a chance to delete it, it would still exist. Those documents would be boxed inside indexes that are created once per `npm run system-test`.